toolchain/parse/handle_binding_pattern.cpp

```diff
@@ -6,46 +6,22 @@
 
 namespace Carbon::Parse {
 
-// Handles BindingPatternAs(ImplicitParam|FunctionParam|Variable|Let).
-static auto HandleBindingPattern(Context& context,
-                                 Context::BindingPatternKind pattern_kind)
-    -> void {
+auto HandleBindingPattern(Context& context) -> void {
   auto state = context.PopState();
 
   // Parameters may have keywords prefixing the pattern. They become the parent
   // for the full BindingPattern.
-  if (pattern_kind != Context::BindingPatternKind::Variable) {
-    context.ConsumeIfBindingPatternKeyword(Lex::TokenKind::Template,
-                                           State::BindingPatternTemplate,
-                                           state.subtree_start);
-    context.ConsumeIfBindingPatternKeyword(Lex::TokenKind::Addr,
-                                           State::BindingPatternAddress,
-                                           state.subtree_start);
-  }
+  context.ConsumeIfBindingPatternKeyword(Lex::TokenKind::Template,
+                                         State::BindingPatternTemplate,
+                                         state.subtree_start);
```

Comment:
Is there a reason that `ConsumeIfBindingPatternKeyword` needs to be a member of `Context`? It looks like it could be defined locally to this file, which is what I think we would generally prefer to do, to keep the interface of `Context` small and keep related functionality together.

Reply:
I agree, but actually I don't think this needs to be a function at all; I'm having a hard time coming up with a documentation comment (or even a name) that seems accurate but doesn't basically spell out the code in words, so I've inlined the function body into the two callsites.